libsolidity/analysis/TypeChecker.cpp

```diff
@@ -1723,6 +1723,8 @@ bool TypeChecker::visit(FunctionCall const& _functionCall)
 			m_errorReporter.typeError(_functionCall.location(), "Type conversion cannot allow named arguments.");
 		else
 		{
+			auto resultArrayType = dynamic_cast<ArrayType const*>(resultType.get());
+			bool isNotAllowedArray = resultArrayType && resultArrayType->isPointer() && !resultArrayType->isByteArray() && !resultArrayType->isString();
```

Comment:
The `isPointer()` should be an assert.

I think that this code should be as close as possible to its usage. Since `isPointer()` is always true, we an actually move it after the re-assignment of `resultType`.